clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -26,22 +37,39 @@
     private final Integer partition;
     private final K key;
     private final V value;
+    private final Long timestamp;
 
     /**
-     * Creates a record to be sent to a specified topic and partition
+     * Creates a record to be sent to a specified topic, partition and timestamp
      * 
      * @param topic The topic the record will be appended to
      * @param partition The partition to which the record should be sent
+     * @param timestamp The timestamp of the record
      * @param key The key that will be included in the record
      * @param value The record contents
      */
-    public ProducerRecord(String topic, Integer partition, K key, V value) {
+    public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
         if (topic == null)
             throw new IllegalArgumentException("Topic cannot be null");
+        if (timestamp != null && timestamp < Record.NO_TIMESTAMP)
```

Comment:
Is it useful for user to specify a -1 timestamp? Would that be the same as passing in a null timestamp?